concrete/concrete.go

```diff
@@ -80,7 +80,8 @@ type PrecompileMap = map[common.Address]Precompile
 type PrecompileRegistry interface {
 	Precompile(address common.Address, blockNumber uint64) (Precompile, bool)
 	Precompiles(blockNumber uint64) PrecompileMap
-	ActivePrecompiles(blockNumber uint64) []common.Address
+	AddPrecompiles(startingBlock uint64, precompiles PrecompileMap)
+	AddPrecompile(startingBlock uint64, address common.Address, precompile Precompile)
```

Comment:
These setters should not be in the interface. The interface and `GenericPrecompileRegistry` are separate things. The interface should only expose getters. But please do add a `PrecompiledAddresses` method that returns a slice of addresses of precompiles for the given block.

Reply:
Noted. will make necessary changes